clients/go/cmd/zbctl/internal/commands/output.go

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2018 Camunda Services GmbH (info@camunda.com)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package commands
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/proto"
+)
+
+type ProtoMessage proto.Message
+type Command cobra.Command
+
+type Printer interface {
```

Comment:
This actually makes sense. I'll try that, thanks :)

Reply:
@MiguelPires Hey! :) I tried to implement methods those return string by each type and I think that better would change them to actually print values. Because some of them used the `log.Println()` function that adds a date for example to output. For JSON output I think that it is not desired. So, I think it is better to keep all print logic inside wrappers. WDYT?

Reply:
Hi @aivinog1. I agree that we don't want to use `log` with the json output but we can use different printing depending on the flag right? Regarding the decision whether to have the commands print or build the string, I would lean towards having the wrappers only return the string. That would separate the concerns and make it easier to change in the future. So the `print` method in output.go would do something like:

 ```go
type Printable interface {
  human() (string, error)
  json() (string, error)
}

func print(p Printable) {
  if humanOutput {
    str, err := p.human()
    // deal with errors
    log.PrintlnIn(str)
  } else if jsonOutput {
    str, err := p.json()
    // deal with errors
    fmt.Println(str)
  }
}

// each command implements Printable and decides what output to return
``` 
This way the commands are only responsible for building the output string and the decision to log or print (for human readable text) is made in a single place which contains all the printing logic. I think that would work right? Let me know if that doesn't make sense or if I misunderstood your point.

Small out of scope detail: One thing that might be worth thinking about also is not even using `log.Println` with human readable text since it's output and not a log statement, so logging it might look strange to an unfamiliar user. But that's out of scope for this issue and if we decide to change it later it should be a small change